lib/vanity/adapters/mock_adapter.rb

```diff
@@ -134,17 +132,27 @@ def ab_not_showing(experiment, identity)
       end
 
       def ab_add_participant(experiment, alternative, identity)
-        @experiments[experiment] ||= {}
-        @experiments[experiment][:alternatives] ||= {}
-        alt = @experiments[experiment][:alternatives][alternative] ||= {}
+        alt = alternative(experiment, alternative)
         alt[:participants] ||= Set.new
         alt[:participants] << identity
       end
 
+      def ab_seen(experiment, identity, alternative)
+        if ab_assigned(experiment, identity) == alternative
+          return alternative
```

Comment:
Yeah, it is a little. If the tests reflect this then it should at least be a bit better documented, I guess. I see from the docs that the adapter API is intended as a public extension point, so I guess we'd have to be pretty careful about "correcting" the method signature now it's out there.